drivers/flash/flash_stm32.c

```diff
@@ -104,6 +107,14 @@ static void flash_stm32_flush_caches(struct device *dev)
 		regs->acr.val &= ~FLASH_ACR_DCRST;
 		regs->acr.val |= FLASH_ACR_DCEN;
 	}
+#else
+	if (regs->acr.val & FLASH_ACR_ARTEN) {
+		regs->acr.val &= ~FLASH_ACR_ARTEN;
+		regs->acr.val |= FLASH_ACR_ARTRST;
+		regs->acr.val &= ~FLASH_ACR_ARTRST;
+		regs->acr.val |= FLASH_ACR_ARTEN;
+	}
+#endif
```

Comment:
This is only valid for access through the ITCM bus. Usually the code is executed through the AXIS bus, and goes through the cache. The data cache should be invalidated through `SCB_InvalidateDCache_by_Addr`. I don't think we should flush the instruction cache after an erase. See https://github.com/aurel32/zephyr/commit/0188becc6d3e91933b6536509a49fe9953fe2acd